### DISCUSSION TITLE
Dynamically change locale without affecting localePath

### DISCUSSION THREAD

**Discussion by nikolajlovenhardt** [nuxt-modules/i18n#2216] [Q&A]
Hi everyone. Thanks for a super cool package. I'm facing a rather complex issue, which I'd like your thoughts on. I doubt if it's possible to solve without forking the repository.

I have an application with individual domains for different languages. I have enabled `differentDomains`. When visitors to the page change language, it redirects to the page that matches the selected language. So far so good!

However, for authenticated users, I'd simply like it to change the language without redirecting to another domain. This is exactly where my challenges arise. Is it possible (maybe with a plugin or middleware) to change the locale without affecting the `localePath`? I've managed to change the locale, but this also changes all the links, which means that whenever an authenticated user presses a link they're redirected to a different domain.

I hope it's alright to post this issue here! Otherwise please let me now. :-)

**Comment by kazupon** [comment]
This is a question, not issue. so we will move to github discussions.

**Comment by kazupon** [comment]
We need a description that includes the actual minimum code so that we can understand your background. 🙏
